packages/mds-audit/types.ts

```diff
@@ -120,13 +122,12 @@ type AuditedDevice =
     })
 // Allow adding type definitions for Express Response objects
 export type AuditApiResponse<B = {}> = ApiVersionedResponse<AUDIT_API_SUPPORTED_VERSION, B> &
-  ApiResponseLocals<ApiClaims<AuditApiAccessTokenScopes>> &
-  ApiResponseLocals<{
-    audit_subject_id: string
-    audit_trip_id: UUID
-    audit: Audit | null
-    recorded: Timestamp
-  }>
+  ApiResponseLocalsClaims &
+  ApiResponseLocalsScopes<AuditApiAccessTokenScopes> &
+  ApiResponseLocals<'audit_subject_id', string> &
+  ApiResponseLocals<'audit_trip_id', UUID> &
+  ApiResponseLocals<'audit', Nullable<Audit>> &
+  ApiResponseLocals<'recorded', Timestamp>
```

Comment:
I wonder if it could be useful to build an abstraction on top of this so we could use the same syntax as before... But could be too tricky to make it worth it. 

I was thinking something like this:
```typescript
type SuperApiResponseLocals<T> = {[U in Extract<keyof T, string>]: ApiResponseLocals<U, T[U]>}
```

Or, perhaps, this could be merged with the new def for ApiResponseLocals so we don't have to have layered types for it (though I prefer composing as opposed to one-lining, especially once it's this complex).

Reply:
Mmm as I dig deeper, this won't work I think. Argh.

Reply:
Yeah, I don’t think there’s any way to map a single object to an intersection of types. I see the next version of Typescript has support for recursive types so maybe there will eventually be support for creating intersections and unions by itereating over properties instead of mapping them.